### PR TITLE
executor: avoid using fmt.Sprint to truncate the string

### LINKED ISSUE #44542: avoid using the fmt.Sprint to truncate the string
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
package main

import (
	"fmt"
)

func main() {
	len1 := int32(1000000)
	len2 := int32(1000001)
	sql := "test"
	fmt.Printf("%.*q", len1, sql)
	fmt.Printf("\n")
	fmt.Printf("%.*q", len2, sql)
}

```

### 2. What did you expect to see? (Required)

normal output
### 3. What did you see instead (Required)

has an error message

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #44542

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by time-and-fate — COMMENTED** [review]
To be compatible with `%q`, which means "a double-quoted string safely escaped with Go syntax", looks like we need to first truncate then `strconv.Quote`.

**Comment by lance6716** [comment]
please add test for the change

File: executor/adapter.go [hunk 1]
```diff
@@ -1247,7 +1247,11 @@ func FormatSQL(sql string) stringutil.StringerFunc {
 			return QueryReplacer.Replace(sql) // no limit
 		}
 		if int32(length) > maxQueryLen {
-			sql = fmt.Sprintf("%.*q(len:%d)", maxQueryLen, sql, length)
+			var result strings.Builder
+			result.Grow(int(maxQueryLen))
+			result.WriteString(fmt.Sprintf("%q", sql)[:maxQueryLen])
+			result.WriteString(fmt.Sprintf("(len:%d)", length))
```
Comment (tiancaiamao) [inline executor/adapter.go:1253]: The fmt.Sprintf seems unnecessary, use `sql[:maxQueryLen]` directly?
And `fmt.Fprintf(result, "(len:%d)", length)`

Comment (hawkingrei) [reply executor/adapter.go]: Yes, I think too. Furthermore, I have asked the original author and they cannot remember the reason for this formatting. Therefore, I have decided to remove this format

**Review by jackysp — APPROVED** [review]
lgtm

**Comment by hawkingrei** [comment]
> To be compatible with `%q`, which means "a double-quoted string safely escaped with Go syntax", looks like we need to first truncate then `strconv.Quote`.

@jackysp  @tiancaiamao and I think it is unnecessary for it in the slow log. so we will remove it.

**Review by xuyifangreeneyes — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**